### PR TITLE
Shots linkage

### PR DESCRIPTION
**Pull request by geenen124**
Link CameraShots to DirectorShots & vice versa. Add methods for generating CameraShots from a DirectorShot and decoupling them. Doesn't work in the GUI due to in-progress DirectorTimeline

### CODE REVIEW

File: src/main/java/gui/modal/CameraShotCreationModalView.java [hunk 1]
```diff
@@ -296,9 +297,10 @@ private void initCamCheckBoxes() {
         // add checkboxes
         cameraCheckboxes = new ArrayList<>();
         int j = 0;
-        for (int i = 0; i < numberOfCameras; i++) {
+        for (int i = 0; i < this.cameraTimelines.size(); i++) {
             j = (j > 4) ? 0 : j + 1;
-            String checkBoxString = "Camera " + (i + 1);
+            String checkBoxString = this.cameraTimelines.get(i).getCamera().getName();
+            System.out.println(checkBoxString);
```
Comment (Markvanderuit) [inline src/main/java/gui/modal/CameraShotCreationModalView.java]: Please remove prints before PR'ing :)

**Comment by Markvanderuit** [comment]
Looks good. Second member can merge.

File: src/main/java/gui/modal/CameraShotCreationModalView.java (hunk 1 above)
Comment (geenen124) [inline src/main/java/gui/modal/CameraShotCreationModalView.java]: Whoops. On it!

**Comment by bart-de-jonge** [comment]
Looks good, please fix the comment posted by @Ezlo123. Then we can merge

**Comment by martijnjanssen** [comment]
Looks good

**Pull request merged**